docs/core-components-1.md

```diff
@@ -0,0 +1,101 @@
+
+* `to`: defines the destination address.
+
+* `sender`: defines the sender address.
+
+* `protocol_id`: defines the id of the protocol.
+
+* `message`: is a bytes field which holds the message in serialized form.
+
+* `Optional[context]`: an optional field to specify routing information in a URI.
+
+Messages must adhere to a protocol.
+
+### Protocol
+
+Protocols define how messages are represented and encoded for transport. They also, optionally, define the rules to which messages have to adhere in a message sequence. 
+
+For instance, a protocol may contain messages of type `START` and `FINISH`. From there, the rules may prescribe that a message of type `FINISH` must be preceded by a message of type `START`.
+
+The <a href="api/protocols/base">Message</a> class in the `protocols/base.py` module provides an abstract class with all the functionality a derived `Protocol` message class requires for a custom protocol, such as basic message generating and management functions and serialisation details.
+
+The framework provides one default protocol, called `default`. This protocol provides a bare bones implementation for an AEA protocol which includes a `DefaultMessage` class and a `DefaultSerialization` class with functions for managing serialisation.
+
+Additional protocols can be added as packages or generated with the <a href="protocol-generator">protocol generator</a>.
+
+Protocol specific messages, wrapped in envelopes, are sent and received to other agents and services via connections.
+
+### Connection
+
+The module `connections/base.py` contains the abstract class which defines a <a href="api/connections/base">Connection</a>. A `Connection` acts as a bridge to the SDK or API to be wrapped, and is, where necessary, responsible for translating between the framework specific `Envelope` with its contained `Message` and the external service or third-party protocol (e.g. `HTTP`).
+
+The framework provides one default connection, called`stub`. It implements an I/O reader and writer to send messages to the agent from a local file. Additional connections can be added as packages.
+
+An AEA can run connections via the `Multiplexer`.
+
+### Multiplexer
+
+The <a href="api/mail/base">Multiplexer</a> is responsible for maintaining potentially multiple connections.
+
+It maintains an `InBox` and `OutBox`, which are, respectively, queues for incoming and outgoing `Envelopes`. They are used to separate the main agent loop from the loop which runs the `Multiplexer`.
+
+### Skill
+
+<a href="api/skill/base">Skills</a> are the core focus of the framework's extensibility. They are self-contained capabilities that AEAs can dynamically take on board, in order to expand their effectiveness in different situations.
+
+A skill encapsulates implementations of the abstract base classes `Handler`, `Behaviour`, `Model` and `Task`:
```

Comment:
the `Task` bit is not technically true anymore.

Reply:
Well you do kind of declare it in or from the skill if you use the packaging approach. Although, I agree it is not part of the skill as such.